test/test.js

```diff
@@ -63,6 +65,27 @@ describe('usb-detection', function() {
 				});
 			});
 
+			it('should return a promise when vid and pid are given', async function(done) {
+				const devices = await usbDetect.find();
+				usbDetect.find(devices[0].vendorId, devices[0].productId)
+					.then(function(devices) {
+						testArrayOfDevicesShape(devices);
```

Comment:
I think we should add these expectations so we know it is actually filtering things

```js
expect(devicesFromTestedFunction.length).to.be.greaterThan(0);
// Should find a subset of USB devices. We assume you have many USB devices connected
expect(devicesFromTestedFunction.length).to.be.lessThan(devices.length);
```

Reply:
I was a bit to hasty on that one sorry, I also called the devices variable the same as the callback. It should be fixed now and thanks for your patience / guidance.